### DISCUSSION TITLE
Order By $raw parameter

### DISCUSSION THREAD

**Discussion by DavideCarvalho** [mikro-orm/mikro-orm#4503] [Ideas]
orderBy: "ASC" or "DESC" looks great for common use cases

But there are some edge cases where we need to do something on ORDER BY clauses, like CAST

I have a case here where the normal "ORDER BY <column>" is ordering wrong, and the solution would be using "ORBER BY column::bytea"

Since we are using repository to query the data, currently I'd need to refactor this whole controller to use knex for this specific scenario, but if we had a $raw, I could only use the raw sql query I want on the order by on this specific column

The idea is to be able to do the following

```ts
this.myRepository.find({}, {
  orderBy: [
    {
      fooColumn: "DESC",
      barColumn: {
        $raw: "bar_column::bytea ASC"
      }
    }
  ]
})
```

Maybe we could also have the following

```ts
this.myRepository.find({}, {
  orderBy: [
    {
      fooColumn: "DESC",
      barColumn: {
        $raw: "bar_column::bytea",
        sort: "ASC"
      }
    }
  ]
})
```

**Comment by B4nan** [comment]
You can already use `raw()` helper in v6, it works for ordering too:

https://github.com/mikro-orm/mikro-orm/blob/v6/docs/docs/raw-queries.md

**Reply by rrrokhtar** [reply]
@B4nan Can you provide an example to use that, since I cannot spot any on docs/raw-queries

**Reply by rrrokhtar** [reply]
Disregard my message managed to resolve it! thanks mate ur effort on that ORM. hats off
For guys who may looking for same of my case, we can workaround FIELD or anything on the orderby query by providing it as a key.
bad thing is u need to hardcode the table alias and column name.
my case:
```js
orderBy: [
 { [raw(`FIELD(\`p0\`.\`id\`, ${ids.join(',')})`)]: 'ASC' },
]
```
